### PR TITLE
Add a github action to run jest, flow, and eslint on PRs

### PR DESCRIPTION
**Pull request by jaredly**
## Summary:
I'm trying out a new "composite action" to take care of setup that will be shared between actions. Looks like it could be useful to us!

Issue: https://khanacademy.atlassian.net/browse/FEI-4334

## Test plan:
see the checks on here...

### CODE REVIEW

File: .github/workflows/pr-checks.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - name: Run ESLint
+      uses: Khan/eslint-action@v2.0.0
+      with:
+        eslint-lib: ./node_modules/eslint
+      env:
+        ALL_CHANGED_FILES: '${{ steps.setup.outputs.added_modified }}'
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Run Jest
+      uses: Khan/jest-action@v1.2.0
```
Comment (somewhatabstract) [inline .github/workflows/pr-checks.yml:44]: I've found these actions to not be as useful as just calling a script from our package.json. In fact in some repos, I've removed them and replaced with those scripts to decrease the number of runtime errors we got from them running (like permissions issues and such). What benefit do we get over just `yarn jest` at this point?

Comment (jaredly) [reply .github/workflows/pr-checks.yml:44]: So these actions give you inline annotation of errors! especially nice for flow & eslint, but also nice for jest imo

Comment (somewhatabstract) [reply .github/workflows/pr-checks.yml:44]: I get that generally with the regular stuff - GitHub has updated significantly and we get the inline eslint stuff even when running it from yarn - and there's a way to customize its matchers to interpret the output for it to auto do this too without maintaining an action for it. Just wondering if we can drop maintenance of these actions in favor of the built-in stuff.

Comment (jaredly) [reply .github/workflows/pr-checks.yml:44]: oh cool! Yeah that would be lovely

File: .github/workflows/pr-checks.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    # edited is needed because that's the trigger when the base branch is
+    # changed on a PR
+    # The rest are the defaults.
+    types: [edited, opened, synchronize, reopened]
+
+jobs:
+  lint_and_test:
+    name: Lint & Test
+    env:
+      CI: true
```
Comment (jeremywiebe) [inline .github/workflows/pr-checks.yml:14]: Doesn't Github set this automatically? https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables

File: .github/workflows/pr-checks.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+name: Lint & Test
+
+on:
+  pull_request:
+    # edited is needed because that's the trigger when the base branch is
```
Comment (jeremywiebe) [inline .github/workflows/pr-checks.yml:5]: Thanks for explaining that!

File: .github/workflows/pr-checks.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        node-version: ${{ matrix.node-version }}
+
+    - name: Run Flow
+      run: yarn flow
+
+    - name: Run ESLint
+      run: yarn eslint .
+
+    - name: Run Jest
+      run: yarn jest --findRelatedTests ${{ steps.setup.outputs.changed_files }}
```
Comment (jeremywiebe) [inline .github/workflows/pr-checks.yml:34]: I'm not super familiar with Github Actions. Will the current directory be the repo this action is used in?

**Review by jeremywiebe — APPROVED** [review]
Cool. Makes sense to me.

File: .github/workflows/pr-checks.yml (hunk 4 above)
Comment (jaredly) [reply .github/workflows/pr-checks.yml:34]: current directory is repo root unless otherwise specified!

**Pull request merged**